UnitySDK/Assets/ML-Agents/Scripts/Agent.cs

```diff
@@ -541,6 +548,7 @@ void ResetData()
                           * param.numStackedVectorObservations]);
 
             info.visualObservations = new List<Texture2D>();
+            info.customOutput = new CustomOutput();
```

Comment:
It is not clear to me how the user is supposed to pass a `customOutput` to the agentInfo. Here you create a new one and the  `GetCustomOutput()` gives you the customOutput in the AgentInfo. I would rather have at this line `info.customOutput = this.customOutput` and have getter and setter on `Agent.customOuptut`.

Reply:
My idea was the user does something like

```cs
void AgentAction(...) {
  var myOutput = GetCustomOutput();
  myOutput.x = 5;
}
```

This was to get around the user having to create their own instance of `CustomOutput` , since then the example becomes

```cs
void AgentAction(...) {
  var myOutput = new CommunicatorObjects.CustomOutput();
  myOutput.x = 5;
  customOutput = myOutput;
```

That seems rather ugly, although it become betters if `CustomOutput` is imported into `MLAgents` namespace.

Are you proposing simply

```cs
void AgentAction(...) {
  customOutput.x = 5;
}
```

with `customOutput` a getter on an internal  attribute that is still created automatically by `Agent` instead of by the user?

Reply:
I would like to avoid creating a new instance of CustomOutput at every step when it is not needed. I am not sure what the API should look like here. To be consistent with AddObservation, we could have a AddCustomOutput method. 
Now that I think about it, having an Agent.customOutput might be confusing. The user might try to modify this value from another script.